### PR TITLE
backend blog

### PR DESCRIPTION
**Pull request by raihankabir36850**
Blog entity create and it's own crud operation

### CODE REVIEW

File: backend/src/controller/blog.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    return res.status(400).json({ errors: errors.array() });
+  }
+
+  const { title, author, description } = req.body;
+  
+  try{
+  // Save to database
+  const BlogsRepository = getConnection().getRepository(Blog);
```
Comment (alamiiin) [inline backend/src/controller/blog.ts]: fix indentation

Comment (alamiiin) [inline backend/src/controller/blog.ts]: blogRepository

File: backend/src/controller/blog.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  const blogs = await BlogsRepository.find({
+    select: ["id","title", "author", "description"],
+  });
+
+  res.json({ data: blogs});
+}
+
+
+// @GET - /api/v1/blogs/
```
Comment (alamiiin) [inline backend/src/controller/blog.ts]: /api/v1/blogs/:blogId

File: backend/src/controller/blog.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    } catch(err) {
+      res.json({err:'Blog is not updated'});
+    }
+    res.json({msg:'Blog is now updated'})
+  }
+
+
+// // @DELETE - /api/v1/blogs/:blogId
+// // update blog
+export async function deletedBlog(req: express.Request, res: express.Response){
```
Comment (alamiiin) [inline backend/src/controller/blog.ts]: deleteBlog

File: backend/src/controller/blog.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    const blogRepository = getConnection().getRepository(Blog);
+    const newBlog = new Blog();
+
+    newBlog.title = title;
+    newBlog.author = author;
+    newBlog.description = description;
+
+    await blogRepository.save(newBlog)
+
+  }catch(err){
```
Comment (farhan2077) [inline backend/src/controller/blog.ts:28]: spacing doesn't match with same try catch block later in this code

File: backend/src/controller/blog.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+      );
+      
+    } catch(err) {
+      res.json({errors:'Blog is not updated'});
+    }
+    res.json({msg:'Blog is now updated'})
+  }
+
+
+// // @DELETE - /api/v1/blogs/:blogId
```
Comment (farhan2077) [inline backend/src/controller/blog.ts]: double comment

File: backend/src/controller/blog.ts [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+// // update blog
+export async function deleteBlog(req: express.Request, res: express.Response){
+    const id=req.params.blogId;
+    const blogRepository= getConnection().getRepository(Blog)
+    try{    
+        if(await blogRepository.delete({id})){
+            return res.json({msg:'delete successfully'})
+        }
+    }catch(err){
+        res.json({errors:'blog is not identified'})
```
Comment (farhan2077) [inline backend/src/controller/blog.ts:92]: Error message can be improved. Ex: "Blog could not be updated."

File: backend/src/controller/blog.ts [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    const newBlog = new Blog();
+
+    newBlog.title = title;
+    newBlog.author = author;
+    newBlog.description = description;
+
+    await blogRepository.save(newBlog)
+
+  }catch(err){
+    res.json({ error: err})
```
Comment (farhan2077) [inline backend/src/controller/blog.ts]: Error message can be improved. Ex: "Blog could not be added."

File: backend/src/controller/index.ts [hunk 8]
```diff
@@ -1,7 +1,21 @@
 export {
-  userLogin as userLoginController,
-  userRegistration as userRegistrationController,
-  users as usersController,
-} from "./auth";
+    userLogin as userLoginController,
```
Comment (farhan2077) [inline backend/src/controller/index.ts]: indentation uses 4 space instead of 2

**Review by alamiiin — APPROVED** [review]

**Pull request merged**